File: server/tcp_server.py
import socket

class TcpServer:
    def __init__(self, port):
        self.host = "127.0.0.1"
        self.port = port

        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
        self.sock.bind((self.host, self.port))

    def listen(self):
        self.sock.listen(1)
        self.client, self.client_address = self.sock.accept()

    def recv(self):
        return self.client.recv(8192)

    def send(self, strs):
        self.client.send(strs.encode('utf-8'))
    
    def close(self):
        self.client.close()

File: 本番/Procon30/TCP_Server.cpp
#include "TCP_Server.h"

void TCP_Server::init(u_short port)
{
	WSAStartup(MAKEWORD(1, 1), &wsaData);
	sock0 = socket(AF_INET, SOCK_STREAM, 0);

	addr.sin_family = AF_INET;
	addr.sin_port = htons(port);
	addr.sin_addr.S_un.S_addr = INADDR_ANY;
	bind(sock0, (struct sockaddr*) & addr, sizeof(addr));
}

void TCP_Server::tcp_open()
{
	listen(sock0, 0);
	len = sizeof(client);
}

bool TCP_Server::tcp_accept()
{
	sock = accept(sock0, (SOCKADDR*)& client, &len);
	return true;
}

void TCP_Server::tcp_send(char *buf, int buflen)
{
	send(sock, buf, buflen, 0);
}

int TCP_Server::tcp_recv(char* buf, int buflen)
{
	memset(buf, 0, buflen);
	return recv(sock, buf, buflen, 0);
}

void TCP_Server::tcp_close()
{
	closesocket(sock);
	WSACleanup();
}